### ISSUE TITLE
Conditional Fields does not work inside a Block

### ISSUE THREAD

**Issue by ozbeksu** [payloadcms/payload#1797]
On the documentation link below, there is a section related to conditional logic for Field types. This works as expected when field is on a Collection type but not when the field is inside a Block type.

My assumption is that since Field type is attachable to both Blocks and Collections, behavior should be same.

https://payloadcms.com/docs/fields/overview#conditional-logic

this works:
```ts
import {CollectionConfig} from 'payload/types';

const Page: CollectionConfig = {
  slug: 'pages',
  fields: [
    // ...
    {
      name: 'enableGreeting', // toggling this checkbox triggers condition logic
      type: 'checkbox',
      defaultValue: false,
    },
    {
      name: 'greeting',
      type: 'text',
      admin: { condition: (data) => data.enableGreeting }
    }
    // ...
  ]
};
```

this doesn't:
```ts
import {Block, CollectionConfig} from 'payload/types';

const GreetingBlock: Block = {
  slug: 'Greeting',
  fields: [
    // ...
    {
      name: 'enableGreeting', // toggling this checkbox should trigger condition logic
      type: 'checkbox',
      defaultValue: false,
    },
    {
      name: 'greeting',
      type: 'text',
      admin: { condition: data => data.enableGreeting },
    },
    // ...
  ],
};

const Page: CollectionConfig = {
  slug: 'pages',
  fields: [
    // ...
    {
      name: 'blocks',
      type: 'blocks',
      minRows: 1,
      maxRows: 20,
      blocks: [GreetingBlock],
    },
    // ...
  ],
};
```

More about my use case (unrelated to bug but may provide some insights):

I have a Pages collection and i'm using this as a route endpoint. Like a Home page, Blog page, Contact page, etc...
I have Blocks. Like Header block, Slider block, Content block, etc...
I also have Blocks containing other Blocks and Relations. Like Header block has CallToAction block, Slider has Media relation, etc...

Basically Pages have a Blocks field where i can mix and match any block type to configure that endpoint's content. Like Home page has a Slider and a Category listing block, or like a Blog Post page has a Header block and a Content block, etc...

On the front-end i'm just looping through the blocks and render components per block to get the page.

**Comment by jmikrut** [comment]
Hey @ozbeksu — this is actually a super common pattern and one that we utilize often.

This is not a bug though and you can indeed do what you're looking to do - the trick is that the `condition` property accepts _**two**_ arguments - the first being the `data` of the entire doc, and then the second being the `siblingData` of the fields _**next**_ to the field that you are applying your condition to.

So, in your case, you'd need to look at the second argument `siblingData`. The docs have a bit more context here but basically you'd need to do this:

```
admin: { condition: (data, siblingData) => siblingData.enableGreeting }
```
